### PR TITLE
Db manager

### PR DESCRIPTION
**Pull request by marioba**
The QWAT db_manager is composed by some parts:
- **Checker** a script that compares 2 databases and shows the differences
- **Dumper** a script to backup and restore a Postgres db
- **Upgrader** a script that apply one or more delta files to an existing database and stores in a table the informations about the applied delta
- **Manager** an interface for managing specific QWAT upgrade procedure

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 language: python
 
+notifications:
+  email: false
+
```
Comment (m-kuhn) [inline .travis.yml]: That probably shouldn't go to production

File: db_manager/dumper.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        command.append('-f')
+        command.append(self.file)
+        command.append(self.__get_dbname())
+
+        try:
+            subprocess.check_output(command, stderr=subprocess.STDOUT)
+
+        except subprocess.CalledProcessError as e:
+            print('pg_dump failed', e.output)
+            raise SystemExit(1)
```
Comment (m-kuhn) [inline db_manager/dumper.py]: That sounds a bit hard, I can imagine we will want to run this from inside QGIS in some user interface the future.

File: system/upgrades_table.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
+-- adds informations table used by the new db_manager
```
Comment (m-kuhn) [inline system/upgrades_table.sql]: "new" will soon no longer be outdated ;)

File: db_manager/test/test_dumper.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        os.mkdir('/tmp/test_dumper/')
+
+    def test_dump_restore(self):
+        dumper = Dumper(self.pg_service1, '/tmp/test_dumper/dump.sql')
+        dumper.pg_backup()
+
+        dumper = Dumper(self.pg_service2, '/tmp/test_dumper/dump.sql')
+        dumper.pg_restore()
+
+        # postgres > 9.4
```
Comment (ponceta) [inline db_manager/test/test_dumper.py]: is postgres strictly > 9.4 or >= 9.4 for this?

**Comment by ponceta** [comment]
Does this has some prerequisites? submodules or external librairies?

What are the specs? Postgres and postgis required and tested versions?

File: .travis.yml (hunk 1 above)
Comment (m-kuhn) [reply .travis.yml]: @marioba can you undo this change?

File: system/upgrades_table.sql (hunk 3 above)
Comment (m-kuhn) [reply system/upgrades_table.sql]: and db_manager should be pum ;)

File: db_manager/test/test_dumper.py (hunk 4 above)
Comment (marioba) [reply db_manager/test/test_dumper.py]: It should be >=9.4 see [here](https://www.postgresql.org/docs/9.6/static/release-9-4.html). 
I used this only in the unit test modules. In all other places, the sql code is >=9.3 as defined in qwat requirements.

File: .travis.yml (hunk 1 above)
Comment (marioba) [reply .travis.yml]: @m-kuhn done!

File: system/upgrades_table.sql (hunk 3 above)
Comment (marioba) [reply system/upgrades_table.sql]: Fixed!

File: db_manager/test/test_dumper.py (hunk 4 above)
Comment (marioba) [reply db_manager/test/test_dumper.py]: It should be >=9.4 see [here](https://www.postgresql.org/docs/9.6/static/release-9-4.html).
I used this only in the unit test modules. In all other places, the sql code is >=9.3 as defined in qwat requirements.

**Comment by m-kuhn** [comment]
Thanks @marioba can you fix the merge conflicts / squash everything?

Next steps:

1. merge this PR
2. integrate view logic
3. fix the deployment of demodata

**Pull request merged**